keyVaultClientSideToCustomerManagedServerSide/setup/setup.py

```diff
@@ -0,0 +1,66 @@
+from azure.storage.blob import BlobServiceClient
+from azure.keyvault.keys import KeyClient
+from azure.identity import ClientSecretCredential
+from azure.keyvault.keys.crypto import CryptographyClient, EncryptionAlgorithm
+from keyVaultClientSideToCustomerManagedServerSide.setup import config as cfg
+
+
+def make_key(k_client):
+    # creates an rsa key by name and saves key to keyvault
+    rsa_key = k_client.create_rsa_key(cfg.keyname, size=2048)  # replace with your keyname
+    k_name = rsa_key.name
+    # return keyvault keyname
+    return k_name
+
+
+def get_key(k_client, k_name):
+    # retrieves key from keyvault by name provided in make_key()
+    keyvault_key = k_client.get_key(k_name)
+    # return keyvault key content
+    return keyvault_key
+
+
+def encrypt_blob(keyvault_key, my_credential):
+    # encrypts blob in bytes using keyvault key from get_key() and credentials from environmental variables
+    message = b"customer managed test"
+    # use credentials and keyvault key to access a client account for encryption
+    crypto_client = CryptographyClient(keyvault_key, credential=my_credential)
```

Comment:
I think client side encryption is a native function of our client, is it not? You should be able to just give it a key and it will wrap the key and encrypt the contents for you

Reply:
Ideally all you will need to do is call upload and it should encrypt and upload for you

Reply:
I think you are right Gauri! 

---
In reply to: [453939666](https://github.com/wastore/azure-storage-samples-for-python/pull/4#discussion_r453939666) [](ancestors = 453939666)